### PR TITLE
fix: graphql getSbomsByLabels and adds unit tests for current queries

### PR DESCRIPTION
**Pull request by gildub**
Add unit tests for all available queries.
Fix `getSbomsByLabels` query to use labels input :
 
image

### CODE REVIEW

File: modules/graphql/src/test.rs [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+        sbom["sha256"],
+        "8f080039c24decc9a066e08fc8f4b7208437536a0bc788d4c76c38c1e1add6e3"
+    );
+
+    log::debug!("{}", data);
+
+    Ok(())
+}
+
+#[ignore]
```
[the remark this answers is not in the document]
Comment (bobmcwhirter) [reply modules/graphql/src/test.rs:221]: I think it's due to more Arc<T> stuff or something not jamming the Database into his context. Intention is to un-ignored, as its failure demonstrates a bug.

**Comment by gildub** [comment]
@bobmcwhirter , with latest grahql's `get_vulnerabilities()` works while `get_advisories()` doesn't.
The later as an issue with db missing in the context, even though `AdvisoryContext` is implemented with lifetime parameter, where `VulnerabilityContext` is not implementing a lifetime parameter.
What's puzzling is that the advisories data is fetched and as same time an error comes back from async-graphql :
`Data `alloc::sync::Arc<trustify_common::db::Database>` does not exist."`

File: modules/graphql/src/endpoints.rs [hunk 2]
```diff
@@ -15,8 +13,7 @@ async fn index_graphiql() -> Result<HttpResponse> {
 
 pub fn configure(svc: &mut web::ServiceConfig, db: Database) {
     let schema = Schema::build(RootQuery::default(), EmptyMutation, EmptySubscription)
-        .data::<Arc<Graph>>(Arc::new(Graph::new(db.clone())))
-        .data::<Arc<Database>>(Arc::new(db))
+        .data::<Database>(db)
```
Comment (bobmcwhirter) [inline modules/graphql/src/endpoints.rs:16]: Does this need to be wrapped back in an Arc<Database>

**Comment by bobmcwhirter** [comment]
> @bobmcwhirter , with latest grahql's `get_vulnerabilities()` works while `get_advisories()` doesn't. The later shows missing db context [1], even though `AdvisoryContext` is implemented with a lifetime parameter, oppositely `VulnerabilityContext` is not.
> 
> What's puzzling advisories data is fetched...[2]
> 
> * [1] `Data `alloc::sync::Arc<trustify_common::db::Database>` does not exist."`
> * [2] https://github.com/trustification/trustify/pull/650/files#diff-a5a9b16d282e20c5882be67f5a32fae9365b7e5e093017ae35e1dbc41391a8bfR45 :
>   image
> 
> So graphql request fails because it cannot reach the context even though the data is retrieved.
> 
> Update : the issue is not occurring when client is not requiring `vulnerabilities` object (Advisory).

VulnContext seems to take an owned clone of Graph, whereas AdvisoryContext is borrowing, which requires a lifetime.

If it's problematic, you can make AdvisoryContext also take an owned clone of Graph.

File: modules/graphql/src/endpoints.rs (hunk 2 above)
Comment (gildub) [reply modules/graphql/src/endpoints.rs:16]: @bobmcwhirter,  I think it does, restoring...

File: modules/graphql/src/test.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (gildub) [reply modules/graphql/src/test.rs:208]: @bobmcwhirter , That solves all context issues !

File: modules/graphql/src/endpoints.rs (hunk 2 above)
Comment (gildub) [reply modules/graphql/src/endpoints.rs]: @bobmcwhirter , That solves all context issues !

**Comment by gildub** [comment]
With latest, only the `get_sbom_by_labels` test failing

**Comment by gildub** [comment]
@bobmcwhirter, @jcrossley3, please re-review.

File: modules/graphql/src/organization.rs [hunk 3]
```diff
@@ -14,7 +15,7 @@ impl OrganizationQuery {
         ctx: &Context<'a>,
         name: String,
     ) -> FieldResult<Organization> {
-        let graph = ctx.data::<Arc<Graph>>()?;
+        let graph: &Arc<Graph> = ctx.data::<Arc<Graph>>()?;
```
Comment (bobmcwhirter) [inline modules/graphql/src/organization.rs:18]: LHS type annotation required?

File: modules/graphql/src/test.rs [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+            id
+            title
+            published
+        }
+    }
+";
+
+#[test_context(TrustifyContext)]
+#[test(tokio::test)]
+async fn get_advisories(ctx: &TrustifyContext) -> Result<(), anyhow::Error> {
```
Comment (bobmcwhirter) [inline modules/graphql/src/test.rs:96]: Nice tests!

File: modules/graphql/src/organization.rs (hunk 3 above)
Comment (gildub) [reply modules/graphql/src/organization.rs:18]: Right, to be removed

File: modules/graphql/src/advisory.rs [hunk 5]
```diff
@@ -11,7 +12,7 @@ pub struct AdvisoryQuery;
 #[Object]
 impl AdvisoryQuery {
     async fn get_advisory_by_id<'a>(&self, ctx: &Context<'a>, id: Uuid) -> FieldResult<Advisory> {
-        let graph = ctx.data::<Arc<Graph>>()?;
+        let graph: &Arc<Graph> = ctx.data::<Arc<Graph>>()?;
```
[the remark this answers is not in the document]
Comment (gildub) [reply modules/graphql/src/advisory.rs:15]: Right, to be removed

File: modules/graphql/src/sbom.rs [hunk 6]
```diff
@@ -35,11 +36,24 @@ impl SbomQuery {
...
         &self,
         ctx: &Context<'a>,
-        labels: Labels,
+        labels: String,
     ) -> FieldResult<Vec<Sbom>> {
-        let graph = ctx.data::<Arc<Graph>>()?;
+        let graph: &Arc<Graph> = ctx.data::<Arc<Graph>>()?;
+        let mut local_labels = Labels::new();
+
+        let labs = labels.split(',');
```
[the remark this answers is not in the document]
Comment (gildub) [reply modules/graphql/src/sbom.rs:44]: @bobmcwhirter, using `insert()` is easier because after using `add()` (from `Labels`) requires cloning.

**Review by bobmcwhirter — APPROVED** [review]

**Pull request merged**